pkg/roachprod/vm/vm.go

```diff
@@ -260,6 +264,17 @@ func (vm *VM) UserName() (string, error) {
 	return parts[0], nil
 }
 
+func (vm *VM) GetCPUArchitecture(l *logger.Logger) (cpuArch CPUArch, _ error) {
+	if err := ForProvider(vm.Provider, func(provider Provider) error {
+		var err error
+		cpuArch, err = provider.GetVMArchitecture(l, vm)
+		return err
+	}); err != nil {
+		return "", err
```

Comment:
Don't we want `vm.ArchUnknown` to distinguish from the empty string?

Reply:
Not sure I understand?

Empty string is returned in case of an error in `GetVMArchitecture()`. Sure, it means the architecture is de facto unknown, but returning empty res+error is the Go way, no?

Reply:
`GetVMArchitecture` can also return `vm.ArchUnknown` when `err` is non-nil. Maybe that's the crux of the issue?